sayanvanish-bukkit/src/main/kotlin/org/sayandev/sayanvanish/bukkit/feature/features/prevent/FeaturePreventCreatureTarget.kt

```diff
@@ -22,14 +24,12 @@ class FeaturePreventCreatureTarget: ListenedFeature("prevent_creature_target", c
         val user = event.user
         val player = user.player() ?: return
         if (StickyNote.isFolia()) {
-            // TODO: Creature modification cannot be off region thread
-            player.world.entities
-                .filterIsInstance<Creature>()
-                .forEach { creature ->
-                    if (creature.target?.uniqueId == player.uniqueId) {
-                        creature.target = null
-                    }
+            player.server.regionScheduler.execute(WrappedStickyNotePlugin.getPlugin().main, player.location) {
+                val entity = Bukkit.getEntity(player.uniqueId)
```

Comment:
In folia you can't receive all entities

Reply:
The current functionality is maintained by the `EntityTargetEvent`:

```kotlin
@EventHandler
private fun onEntityTarget(event: EntityTargetEvent) {
    if (!isActive()) return
    val target = event.target as? Player ?: return
    if (!SayanVanishBukkitAPI.getInstance().isVanished(target.uniqueId)) return
    event.isCancelled = true
}
```

The runnable is just an additional layer. In the current code, the following part:

```kotlin
player.server.regionScheduler.execute(WrappedStickyNotePlugin.getPlugin().main, player.location) {
    val entity = Bukkit.getEntity(player.uniqueId)
```

doesn't actually do anything. It sets the player's target to null, not the entities that target the player. I believe removing the runnable part entirely should be fine, though I kept it just in case. It might be better to remove that part entirely.